### ISSUE TITLE
Error: Invalid address to set

### ISSUE THREAD

**Issue by joby-kurian** [terraform-google-modules/terraform-google-kubernetes-engine#1544]
### TL;DR

I am using GKE and ASM module for my development since august 2022. I was able to deploy GKE and ASM using below repo
"terraform-google-modules/kubernetes-engine/google//modules/asm" and version = "20.0.0". But today I am getting one strange error, even without any change from my end. I tried with old branch which run successfully, also giving this error.

Error: Invalid address to set: []string{"addons_config", "0", "config_connector_config"}
│ 
│   with module.asm.module.asm.data.google_container_cluster.asm,
│   on .terraform/modules/asm.asm/modules/asm/main.tf line 28, in data "google_container_cluster" "asm":
│   28: data "google_container_cluster" "asm" {

### Expected behavior

The code should run without any error.

### Observed behavior

But it's throwing "Error: Invalid address to set". 

Error: Invalid address to set: []string{"addons_config", "0", "config_connector_config"}
│ 
│   with module.asm.module.asm.data.google_container_cluster.asm,
│   on .terraform/modules/asm.asm/modules/asm/main.tf line 28, in data "google_container_cluster" "asm":
│   28: data "google_container_cluster" "asm" {

### Terraform Configuration

```hcl
module "asm" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version = "20.0.0"

  project_id                = var.project_id
  cluster_name              = var.cluster_name
  cluster_location          = var.cluster_location
  enable_cni                = true
  enable_fleet_registration = true
  enable_mesh_feature       = true
  enable_vpc_sc             = true
  fleet_id                  = var.fleet_name
}
```

### Terraform Version

```sh
Terraform v1.3.3

google = {
      source  = "hashicorp/google"
      version = ">= 4.20.0"
    }
    vault = {
      source  = "hashicorp/vault"
      version = "2.23.0"
    }
```

### Additional information

Since last week I was able to run the code without any issues, not sure what happened now.

**Comment by joby-kurian** [comment]
As you suggested, it is working after google provider downgrade. Thank you so much for the update.

**Comment by bmistrapau** [comment]
Hello,

We are facing the same bug described above, when using google provider v4.50.0 and gke terraform module v23.1.0 :

```
Error: Invalid address to set: []string{"addons_config", "0", "config_connector_config"}

  with module.oro-gke-private-cluster01.module.acm[0].module.registration.data.google_container_cluster.primary,
  on .terraform/modules/oro-gke-private-cluster01.acm.registration/modules/local/terraform-google-modules/kubernetes-engine/google/v23.1.0/modules/fleet-membership/main.tf line 23, in data "google_container_cluster" "primary":
  23: data "google_container_cluster" "primary" {


Error: Invalid address to set: []string{"addons_config", "0", "config_connector_config"}

  with module.oro-gke-private-cluster01.module.asm[0].data.google_container_cluster.asm,
  on .terraform/modules/oro-gke-private-cluster01.asm/modules/local/terraform-google-modules/kubernetes-engine/google/v23.1.0/modules/asm/main.tf line 28, in data "google_container_cluster" "asm":
  28: data "google_container_cluster" "asm" {
```


The terraform code referenced is:


```
# Retrieve GKE cluster info
data "google_container_cluster" "primary" {
  name     = var.cluster_name
  location = var.location
  project  = var.project_id
}
```

and

```
data "google_container_cluster" "asm" {
  project  = var.project_id
  name     = var.cluster_name
  location = var.cluster_location

  depends_on = [var.module_depends_on]
}
```

Thank you,
Bogdan
